CODE_REVIEW.md

```diff
@@ -0,0 +1,23 @@
+# Improvements:
+
+* Website is not responsive for mobile devices, made it responsive.
+* Private member variable ‘http' is never reassigned, marked it as 'readonly’. Added readonly (private readonly actions$: Actions, private readonly http: HttpClient) for actions$ and http in readling-list.effects.ts file.
+* Filtering reading list API (/api/reading-list) response and passing only the list with book items, thereby avoiding any empty objects.
+* this.store.select(getAllBooks) value is not unsubscribing after Subscription in book-searcg.Component.ts file may result. If a subscription is not closed the function callback attached to it will be continuously called, this may poses a huge memory leak and performance issue. Added async pipe to avoid potential memory leaks.
+* When user searches for a keyword, we can use a service to cache the results of /api/books/search?q=${action.term} api response and retrieve the data from cache if he searches for same keyword again thereby avoid hitting same endpoint multiple times. Since reading the data from cache is extremely fast, this improves overall performance of the application.
+* We can add routing for search key value. Even if user reloads the page, he can land on the search item.
```

Comment:
Explain broadly how you might do this?

Reply:
When we show the list of books for the keyword user had searched(Ex: **javascript**), we will add that keyword in the query params(**?bookSearched=javascript**). So, if the user reloads the page, we can make him stay on the same page using the book name in query params and show in related content by calling api(**/api/books/search?q=javascript**).

Reply:
Not what I was asking. How _in Angular_ would you do this?

Reply:
On load of the component, check if the query parameter exists in the URL or not. If it exists run the same action as if they searched it. We also need to update the url query parameter after every book search.